src/ai/backend/agent/stats.py

```diff
@@ -266,7 +281,7 @@ def __init__(
         self, agent: "AbstractAgent", mode: StatModes = None, *, cache_lifespan: int = 120
     ) -> None:
         self.agent = agent
-        self.mode = mode if mode is not None else StatModes.get_preferred_mode()
+        self.mode = mode if mode == StatModes.DOCKER else StatModes.get_preferred_mode()
```

Comment:
The argument is optional and `None` means that the caller is leaving the choice of mode to the implementation. So, this should not be changed.